File: 2.0/java/texts/findTextsNonProfitAgency.java
import com.callfire.api.client.CallfireClient;
import com.callfire.api.client.api.callstexts.model.Text;
import com.callfire.api.client.api.callstexts.model.request.FindTextsRequest;
import com.callfire.api.client.api.common.model.Page;

class ApiClientSample {
    public static void main(String[] args) {
        CallfireClient client = new CallfireClient("api_login", "api_password");
        FindTextsRequest request = FindTextsRequest.create()
            // find messages sent from particular campaign or replies.
            .campaignId(1003002L)
            // find only inbound messages
            .inbound(true)
            .limit(10L)
            .build();
        Page<Text> texts = client.textsApi().find(request);
    }
}

File: 2.0/csharp/texts/findTextsNonProfitAgency.cs
using CallfireApiClient;
using CallfireApiClient.Api.CallsTexts.Model;
using CallfireApiClient.Api.CallsTexts.Model.Request;
using CallfireApiClient.Api.Common.Model;

public class ApiClientSample
{
    public static void Main(string[] args)
    {
        var client = new CallfireClient("api_login", "api_password");
        var request = new FindTextsRequest
        {
            // find all replies for a particular Text Broadcast
            CampaignId = 1003003,
            Inbound = true,
            Limit = 10
        };
        Page<Text> texts = client.TextsApi.Find(request);
    }
}
